### PR TITLE
[API-789] Fix async unit tests

### PR DESCRIPTION
**Pull request by elseee**
https://bynder.atlassian.net/browse/API-789

- Fixed some async tests which were not waiting for the promise to complete
- Fixed the issues which arose from some tests that turned out to be false positives

### CODE REVIEW

File: tests/src/index.test.js [hunk 1]
```diff
@@ -131,77 +129,65 @@ describe('#getToken', () => {
 
 describe('#uploadFile', () => {
   it('throws an error with no brand ID', () => {
-    bynder.uploadFile({
+    return expect(bynder.uploadFile({
```
Comment (elseee) [inline tests/src/index.test.js:132]: To make sure that jest waits for the promise to complete we need to either return the promise from the test, or await it. Also by the `expect(...).rejects` pattern we ensure that the test fails if the promise were to resolve.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [ 12.x, 13.x, 14.x, 15.x ]
```
Comment (betacar) [inline .github/workflows/run-tests.yml:16]: Please add the current LTS: `16.x`.

Comment (elseee) [reply .github/workflows/run-tests.yml:16]: Done!

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [ 12.x, 13.x, 14.x, 15.x ]
```
[the remark this answers is not in the document]
Comment (elseee) [reply .github/workflows/publish.yml:16]: Done!

File: src/index.js [hunk 4]
```diff
@@ -644,11 +644,17 @@ class Bynder {
...
    * @param {String} data.mediaId Media ID
+   * @param {String} data.brandId Brand ID
    * @param {String} data.name Asset name
    * @return {Promise<object>}
    */
   _saveAsset(data) {
-    const { fileId, mediaId, additional } = data;
+    const { fileId, mediaId, additional, brandId } = data;
+
+    if (!brandId && !mediaId) {
```
[the remark this answers is not in the document]
Comment (betacar) [reply src/index.js:654]: But this condition is requesting both parameters to be defined.

[the remark this answers is not in the document]
Comment (elseee) [reply src/index.js:654]: The condition is checking if both of them are undefined, hence the `!`

**Review by betacar — APPROVED** [review]

**Pull request merged**